tests/testthat/test-fimsframe.R

```diff
@@ -208,4 +208,33 @@ test_that("get_n_fleets() works with correct inputs", {
 # No edge cases to test.
 
 ## Error handling ----
-# No built-in errors to test.
+
+## Check that user does not include unsupported data types
+test_that("FIMSFrame() aborts on unsupported data types", {
+  data("data1", package = "FIMS")
+  bad <- dplyr::mutate(
+    data1,
+    type = ifelse(type == "index", "indexes", type)  # Introduce an unsupported type
+  )
+  expect_error(FIMS::FIMSFrame(bad), regexp = "unsupported type\\(s\\)")
+})
+
+##Check that a warning, not an error, is thrown when an expected type is missing
+test_that("FIMSFrame() warns but constructs when an expected type is missing", {
+  data("data1", package = "FIMS")
+  # Removes length_comp
+  no_len <- dplyr::filter(data1, type != "length_comp")
+
+  expect_warning(
+    ff <- FIMS::FIMSFrame(no_len),
+    regexp = "expected data type\\(s\\) are not present"
+  )
+  expect_s4_class(ff, "FIMSFrame")
+})
+
+##Confirm that valid inputs work without error
+test_that("FIMSFrame() succeeds cleanly with valid inputs", {
```

Comment:
Thanks for writing this test! Do you think we could move it to the IO correctness section, maybe around line 25?